### PR TITLE
Support dotnet-isolated runtime when deploying to KEDA

### LINKED ISSUE #2825: Azure Functions Core Tools v4 unable deploy to Kubernetes
```txt
G:\Projects\funapp2>func --version
4.0.3928

G:\Projects\funapp2>dotnet --version
6.0.100
```

Here are the steps to reproduce the problem.

```sh
mkdir funapp2 && cd funapp2
func init --worker-runtime dotnetIsolated --docker
func new -t "HttpTrigger" -n "Login" -a "anonymous"
```

Modify the `Dockerfile` file as below:

```dockerfile
FROM mcr.microsoft.com/dotnet/sdk:6.0 AS installer-env

COPY . /src/dotnet-function-app
RUN cd /src/dotnet-function-app && \
    mkdir -p /home/site/wwwroot && \
    dotnet publish *.csproj --output /home/site/wwwroot

# To enable ssh & remote debugging on app service change the base image to the one below
# FROM mcr.microsoft.com/azure-functions/dotnet-isolated:3.0-dotnet-isolated5.0-appservice
FROM mcr.microsoft.com/azure-functions/dotnet-isolated:3.0-dotnet-isolated5.0
ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
    AzureFunctionsJobHost__Logging__Console__IsEnabled=true

COPY --from=installer-env ["/home/site/wwwroot", "/home/site/wwwroot"]
```

Build & Push image

```sh
docker build -t willh/funapp2:latest .
docker push willh/funapp2:latest
```

Deploy my function app with `--dry-run`

```sh
func kubernetes deploy --name funapp2 --service-type NodePort --image-name willh/funapp2 --dry-run
```

There is only **Secret** been created.

```sh
data:
  AzureWebJobsStorage: VXNlRGV2ZWxvcG1lbnRTdG9yYWdlPXRydWU=
  FUNCTIONS_WORKER_RUNTIME: ZG90bmV0LWlzb2xhdGVk
apiVersion: v1
kind: Secret
metadata:
  name: funapp2
---
```

Where is the other YAML of the deployment?

### PR DESCRIPTION
**Pull request by f1x3d**
### Issue describing the changes in this PR

Resolves #2825

Apparently, the Kubernetes deployment part of the function tools was not updated to support `dotnet-isolated` worker runtime since it still looks for `function.json` files which [are not being generated with the new runtime](https://docs.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#differences-with-net-class-library-functions).

That being said, the functions metadata still gets created in the `functions.metadata` file which is essentially an array of objects with similar to `function.json` structure.

This PR adds support of `dotnet-isolated` runtime functions by reading the metadata from the new `functions.metadata` file.

To test this, create a new `dotnet-isolated` function project with at least one trigger and run `func kubernetes deploy --name testname --registry testregistry --dry-run --dotnet-isolated`
The output should contain yamls for `Deployment` and `ScaledObject` with corresponding data.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [x] Otherwise: Backport tracked by issue/PR #2556
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by f1x3d** [comment]
Also, PR checklist mentions adding unit/e2e tests but the [contributing guide](https://github.com/Azure/azure-functions-core-tools/blob/v4.x/CONTRIBUTING.md#running-the-test-suite) says you can't run tests at this moment, so I would need some help with that

**Comment by jia-ming-li-rs** [comment]
Hi @AnatoliB @michaelpeng36  Azure Function (none-isolated) with KEDA is broken on .NET 7. please help get this dotnet-isolated to work. thank you.

**Comment by michaelpeng36** [comment]
@jia-ming-li-rs Thank you for the ping. I am no longer working on Core Tools deployments. @FinVamp1, could you take a look at this?

**Comment by FinVamp1** [comment]
Thanks @michaelpeng36 I am looking at this. I think we want to test with larger numbers of functions to make sure it scales and it would be good to have some unit and\or e2e tests. Adding @khkh-ms  to help.

**Review by justinmchase — APPROVED** [review]

**Comment by claudio-cauchi** [comment]
@FinVamp1 @khkh-ms We are also blocked on this PR. I'm not sure what tests need to be added as the contributing.md file states that the tests are not working on dotnetcore.

**Review by khkh-ms — APPROVED** [review]

**Comment by khkh-ms** [comment]
The PR couldn't be merged because of some build issue which fails the build from forks. We are working to resolve that issue. However, to unblock this change, I have created a different PR with the commit of this PR and have merged it. Thanks, you!

**Pull request closed without merging**
